src/Material.cpp

```diff
@@ -1518,3 +1619,230 @@ Material* Material::clone(){
 
   return clone;
 }
+
+
+/* ************************************************************************** */
+
+
+/**
+ * @brief Constructor sets the ID and unique ID for the Material.
+ * @param id the user-specified optional Material ID
+ * @param name the user-specified optional Material name
+ */
+IsoMaterial::IsoMaterial(int id, const char* name) : Material(id,name) {
+    
+  _material_type = ISO_MATERIAL;
+  
+  _num_isotopes = 0;
+  
+  return;
+}
+
+
+/**
+ * @brief Destructor: clears dynamically allocated memory.
+ */
+IsoMaterial::~IsoMaterial() {
+  
+  _isotopes.clear();
+  _num_dens.clear();
+  
+  return;
+}
+
+
+/**
+ * @brief Add an isotope to the Material.
+ * @details Add the isotope's contributions to the stored cross sections
+ * @param isotope Pointer to the Isotope to be added
+ * @param number_density the number density of the Isotope in the Material
+ */  
+void IsoMaterial:: addIsotope(Isotope* isotope, FP_PRECISION number_density) {
+  int num_groups = isotope->getNumEnergyGroups();
+  bool chi_set = false;
+  
+  if (_num_groups==0) {
+    setNumEnergyGroups(num_groups);
+    _chi_set = false;
+  }
+  if (num_groups != _num_groups)
+    log_printf(ERROR,"Cannot add Isotope %d because it has %d energy groups "
+               "and other isotopes in Material have %d.", isotope->getId(),
+               num_groups, _num_groups);
+  
+  _isotopes.push_back(isotope);
+  _num_dens.push_back(number_density);
+  
+  for (int g=0; g<_num_groups; g++) {
+    _sigma_t[g] += isotope->getSigmaTByGroup(g+1)*number_density;
+    _sigma_a[g] += isotope->getSigmaAByGroup(g+1)*number_density;
+    _sigma_f[g] += isotope->getSigmaFByGroup(g+1)*number_density;
+    _nu_sigma_f[g] += isotope->getNuSigmaFByGroup(g+1)*number_density;
+    if (not _chi_set) {
+      _chi[g] = isotope->getChiByGroup(g+1);
+      if (_chi[g] > 0) chi_set = true;
+    }
+  }
+  
+  if (chi_set) _chi_set = true;
+  
+  if (isotope->isFissionable())
+    _fissionable = true;
+  
+  _num_isotopes++;
+}
+
+
+/**
+ * @brief Get specific number density from Material
+ * @param index The index of the material to be retrieved
```

Comment:
I think this is a typo: "material" -> "isotope". Also, the isotope isn't "retrieved", only its number density is, correct?